### ISSUE TITLE
bug(YouTube): app crashes when `SponsorBlock` patch is excluded

### ISSUE THREAD

**Issue by NoMorePizza** [inotia00/ReVanced_Extended#2068]
### Type

Error at runtime

### Tools used

RVX Manager

### Application

YouTube v19.20.34
Patches v4.8.3

### Bug description

When I try to open the app, it just crashes and closes on startup. 

### Additional context

Sorry for not having any error logs, I don't know how to get them. I'm not tech-savvy enough.

### Device Environment

Android 14, OneUI 6.0

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
Since there is no log, there is no context I can provide.
Try patching with YouTube v19.16.39

**Comment by NoMorePizza** [comment]
I patched with v19.16.39 and now the app opens normally. I wonder why v19.20.34 doesn't work.

Is it difficult to get crash logs? Do I need a rooted device for that?

**Comment by NoMorePizza** [comment]
O

**Comment by inotia00** [comment]
> Is it difficult to get crash logs? Do I need a rooted device for that?

I assume [termux](https://github.com/KitsunedFox/termux-monet/releases/latest) is installed on your device.

1. Open termux and enter the following command:
```
logcat | grep AndroidRuntime > /sdcard/log.txt
```
2. Open RVX and reproduce the force close
3. Go back to termux and type CTRL + C
4. Attach `log.txt`

**Comment by NoMorePizza** [comment]
I tried without any success. Do I need an SD card for that? Because mine recently bricked itself.

**Comment by inotia00** [comment]
nah `/sdcard` means internal storage

**Comment by ILoveOpenSourceApplications** [comment]
I'm also having the same issue and would like to contribute. Is the command  for Windows the one given below?
```
adb logcat | findstr AndoridRuntime > logcat.txt
```

**Comment by inotia00** [comment]
yes

**Comment by ILoveOpenSourceApplications** [comment]
```kotlin
05-26 21:09:15.616 31259 31259 D AndroidRuntime: Shutting down VM
05-26 21:09:15.617 31259 31259 E AndroidRuntime: FATAL EXCEPTION: main
05-26 21:09:15.617 31259 31259 E AndroidRuntime: Process: app.rvx.android.youtube, PID: 31259
05-26 21:09:15.617 31259 31259 E AndroidRuntime: java.lang.VerifyError: Verifier rejected class kxr: void kxr.F() failed to verify: void kxr.F(): [0x40B] register v7 has type Reference: android.widget.LinearLayout but expected Reference: android.widget.FrameLayout (declaration of 'kxr' appears in /data/app/~~xjmdOsTUENvhPOoH1fwVcA==/app.rvx.android.youtube-GvYsWoWw2qGAup4YSwQMVQ==/base.apk!classes6.dex)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at kxa.v(PG:1)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.b(PG:8)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.a(PG:88)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azol.a(PG:16)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azok.a(PG:5)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.l(PG:63)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.a(PG:73)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azol.a(PG:16)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.l(PG:64)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.a(PG:73)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azol.a(PG:16)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xnk.b(PG:3)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xnk.<init>(PG:10)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at nmg.<init>(PG:18)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at nlj.t(PG:1)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.k(PG:7)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at gbm.a(PG:78)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azol.a(PG:16)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at azok.a(PG:5)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xnk.b(PG:3)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xnk.a(PG:5)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at njt.run(PG:256)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xxq.run(PG:20)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xxs.l(PG:59)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at xxs.m(PG:57)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at com.google.android.apps.youtube.app.watchwhile.MainActivity.onCreate(PG:983)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8975)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:8944)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1456)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4146)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4322)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:230)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:319)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8919)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
05-26 21:09:15.617 31259 31259 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

**Comment by inotia00** [comment]
Thanks for providing the logs
I'll reopen the issue

**Comment by inotia00** [comment]
If possible, can you export settings and upload here?

**Comment by inotia00** [comment]
Also please specify which YouTube version the crash log is from

**Comment by ILoveOpenSourceApplications** [comment]
> If possible, can you export settings and upload here?

I'm unable to open the app as the crash is instantaneous and I don't have any backup either for my YouTube settings. So I'm pretty much in a pickle. If I'm to uninstall and install v19.16.39, I'll have to toggle each setting one by one.

**Comment by inotia00** [comment]
This seems to be the crash log of YouTube v19.21.34 (unsupported version)

**Comment by ILoveOpenSourceApplications** [comment]
> Also please specify which YouTube version the crash log is from

Log is from v19.21.34. I know the recommended version is v19.20.34 but I misread the issue #2070 and somehow came to the conclusion that the app opens in that version and hence I thought maybe I  can open the app and import the settings if the app opens. But i was wrong.

**Comment by ILoveOpenSourceApplications** [comment]
I can open the app with basic patches such as `Hide ads, GmsCore support, Theme, etc`. But if I add anything more the app starts crashing.

**Comment by ILoveOpenSourceApplications** [comment]
I can provide my patch selection from Manager at best now or I'll have to downgrade and change the settings once again.

**Comment by inotia00** [comment]
Please attach crash log from YouTube v19.20.34

**Comment by ILoveOpenSourceApplications** [comment]
```
05-27 00:29:56.791 16130 16130 E AndroidRuntime: FATAL EXCEPTION: main
05-27 00:29:56.791 16130 16130 E AndroidRuntime: Process: app.rvx.android.youtub, PID: 16130
05-27 00:29:56.791 16130 16130 E AndroidRuntime: baif: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.lang.ExceptionInInitializerError
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at aztl.aa(PG:39)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at bahp.run(PG:8)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:958)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:230)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:319)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8919)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
05-27 00:29:56.791 16130 16130 E AndroidRuntime: Caused by: java.lang.ExceptionInInitializerError
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at app.revanced.integrations.youtube.patches.video.PlaybackSpeedPatch.newVideoStarted(PlaybackSpeedPatch.java:31)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at gzv.setVideoInformation(Unknown Source:27)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at gzv.l(PG:35)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at gwx.a(PG:92)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at bbic.wZ(PG:9)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at barp.wZ(PG:3)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at barm.d(PG:113)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at barm.wZ(PG:24)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at baqr.g(PG:13)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at baqp.run(PG:5)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at azvw.run(PG:27)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at bahp.run(PG:3)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        ... 8 more
05-27 00:29:56.791 16130 16130 E AndroidRuntime: Caused by: android.content.res.Resources$NotFoundException: Resource ID #0x0
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:270)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.content.res.Resources.getDrawableForDensity(Resources.java:1049)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.content.res.Resources.getDrawable(Resources.java:989)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at android.content.res.Resources.getDrawable(Resources.java:964)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at app.revanced.integrations.shared.utils.ResourceUtils.getDrawable(ResourceUtils.java:103)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        at app.revanced.integrations.youtube.whitelist.Whitelist.<clinit>(Whitelist.java:50)
05-27 00:29:56.791 16130 16130 E AndroidRuntime:        ... 20 more
```

**Comment by ILoveOpenSourceApplications** [comment]
The two patches which are causing the issue for me are `Fullscreen components` and `Video playback`.

**Comment by inotia00** [comment]
Thanks for providing the logs
Could you please provide the YouTube version used for the patch and a full list of patches to include or exclude?

**Comment by ILoveOpenSourceApplications** [comment]
> Thanks for providing the logs ~Could you please provide the YouTube version used for the patch and a full list of patches to include or exclude?~

The version I used for the logs is v19.20.34.

> nvm, I found the cause of the issue Thank you again for finding out which patch is causing the issue

Oh, I see. So it's not caused by those patches but because of excluding `SponsorBlock`.

**Comment by inotia00** [comment]
fixed in [revanced-patches-4.8.4](https://github.com/inotia00/revanced-patches/releases/tag/v4.8.4)
